helpers/integration-tests.js

```diff
@@ -0,0 +1,53 @@
+const puppeteer = require('puppeteer');
+
+const URLsToVerify = [
+    'https://sonarwhal-staging.azurewebsites.net/',
+    'https://sonarwhal-staging.azurewebsites.net/scanner/',
+    'https://sonarwhal-staging.azurewebsites.net/search/?q=bla',
+    'https://sonarwhal-staging.azurewebsites.net/about/changelog/1',
+    'https://sonarwhal-staging.azurewebsites.net/docs/user-guide/hints/'
+];
+
+puppeteer.launch()
+    .then((browser) => {
+        const responseCollection = new Promise((resolve) => {
+            const responses = new Map();
+
+            URLsToVerify.forEach((url, index) => {
+                browser.newPage()
+                    .then((page) => {
+                        page.goto(url)
+                            .then((response) => {
+                                responses.set(url, response.status());
+                            })
+                            .catch((error) => {
+                                responses.set(url, `❌ ${error}`);
+                            })
+                            .finally(() => {
+                                if (index === (URLsToVerify.length - 1)) {
+                                    resolve(responses);
+                                    browser.close();
```

Comment:
[`browser.close()` returns a Promise](https://pptr.dev/#?product=Puppeteer&version=v1.20.0&show=api-pagecloseoptions).
I'm thinking of 2 possibilities, if you have a better one please go for it:

1. Add the condition after `responses.set()`. Although we might end up with a bit more nesting 🤔
1. Use `async/await` inside a `for..of` and test the pages one by one. This might make the code a bit easier even if it means the test will take a few more seconds (which is fine TBH). 

What do you think?

Reply:
I don't think we need to keep the responses in a `Map`.
As @molant says, you should use `async/await` inside a `for..of` and after each scan, print the status (that way we can see if it is failing more than one url) and after all the urls are scanned, throw and error if there was any exception (or maybe it is enough if we return an exitCode !== 0)